src/chrome/chromeDebugAdapter.ts

```diff
@@ -646,75 +655,84 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
-            if (this._hasTerminated) {
-                return undefined;
-            }
+                if (sources) {
+                    const filteredSources = sources.filter(source => source !== mappedUrl); // Tools like babel-register will produce sources with the same path as the generated script
+                    for (const filteredSource of filteredSources) {
+                        await resolvePendingBPs(filteredSource);
+                    }
+                }
 
-            if (sources) {
-                // If break on load is active, check whether we should call resolvePendingBPs
-                if (this.breakOnLoadActive) {
-                    sources
-                        .filter(source => source !== mappedUrl && this._breakOnLoadHelper.shouldResolvePendingBPs(source)) // Tools like babel-register will produce sources with the same path as the generated script
-                        .forEach(resolvePendingBPs);
+                if (script.url === mappedUrl && this._pendingBreakpointsByUrl.has(mappedUrl) && this._pendingBreakpointsByUrl.get(mappedUrl).bpsSet) {
+                    // If the pathTransformer had no effect, and we attempted to set the BPs with that path earlier, then assume that they are about
+                    // to be resolved in this loaded script, and remove the pendingBP.
+                    this._pendingBreakpointsByUrl.delete(mappedUrl);
                 } else {
-                    sources
-                        .filter(source => source !== mappedUrl) // Tools like babel-register will produce sources with the same path as the generated script
-                        .forEach(resolvePendingBPs);
+                    await resolvePendingBPs(mappedUrl);
                 }
-            }
 
-            if (script.url === mappedUrl && this._pendingBreakpointsByUrl.has(mappedUrl) && this._pendingBreakpointsByUrl.get(mappedUrl).bpsSet) {
-                // If the pathTransformer had no effect, and we attempted to set the BPs with that path earlier, then assume that they are about
-                // to be resolved in this loaded script, and remove the pendingBP.
-                this._pendingBreakpointsByUrl.delete(mappedUrl);
-            } else {
-                // If break on load is active, check whether we should call resolvePendingBPs
-                if (!this.breakOnLoadActive || (this._breakOnLoadHelper && !sources && this._breakOnLoadHelper.shouldResolvePendingBPs(mappedUrl))) {
-                    resolvePendingBPs(mappedUrl);
-                }
-            }
+                await this.resolveSkipFiles(script, mappedUrl, sources);
+            });
 
-            return this.resolveSkipFiles(script, mappedUrl, sources);
-        });
+            if (this._initialSourceMapsP) {
+                this._initialSourceMapsP = <Promise<any>>Promise.all([this._initialSourceMapsP, sourceMapsP]);
+                await sourceMapsP;
```

Comment:
Should this always be awaited outside the if block? And why does it need to be?

Reply:
You are right, it should be outside.

We want this so we don't call breakpointsAreResolvedDefer.resolve(); until this has finished for sure. If we call breakpointsAreResolvedDefer.resolve(); too early that will cause issues.